tests/test_modeling_common.py

```diff
@@ -245,15 +245,22 @@ def _create_and_check_torchscript(self, config, inputs_dict):
             inputs = self._prepare_for_class(inputs_dict, model_class)["input_ids"]  # Let's keep only input_ids
 
             try:
-                traced_gpt2 = torch.jit.trace(model, inputs)
+                if model.config.is_encoder_decoder:
+                    # Bart cannot trace decoder cache for now.
+                    # see: https://github.com/huggingface/transformers/issues/6348
+                    # TODO: Remove following line when issue is fixed.
+                    model.config.use_cache = False
+                    traced_model = torch.jit.trace(model, (inputs, inputs))
+                else:
+                    traced_model = torch.jit.trace(model, inputs)
             except RuntimeError:
                 self.fail("Couldn't trace module.")
 
             with tempfile.TemporaryDirectory() as tmp_dir_name:
                 pt_file_name = os.path.join(tmp_dir_name, "traced_model.pt")
 
                 try:
-                    torch.jit.save(traced_gpt2, pt_file_name)
+                    torch.jit.save(traced_model, pt_file_name)
```

Comment:
(out of scope) 
Why is the try/except -> `self.fail` pattern useful?

Without try/except you get an error that traces back to a line with the word `save` in it.
I just looked at `self.fail` and it turns things into AssertionErrors, which seems like strictly less info.